### ISSUE TITLE
ScrollTrigger feature request: allow passing in an existing spacer element

### ISSUE THREAD

**Issue by cubeghost** [greensock/GSAP#461]
Currently, pinning an element with ScrollTrigger wraps it in a spacer element to keep the page layout intact while the pinned element is `position: fixed;`. Because this always adds an ancestor to the DOM tree, it's impossible to pin an element that has `iframe` children without [the frames reloading](https://greensock.com/forums/topic/26418-iframe-in-pinned-element-reloading-on-resize-event/?tab=comments#comment-128558).

Manipulating the DOM tree this way also doesn't play well with React- if the sibling of a pinned element needs to be added or removed from the DOM by React's internals, it often crashes with the message `Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.`.

One solution to both of these problems would be to allow users to pass in an existing wrapper/spacer element (as a selector or an actual element reference), like below:

```html
<div class="custom-spacer">
  <div class="pin-me"></div>
</div>
```
```js
ScrollTrigger.create({
  trigger: ".pin-me",
  pin: true,
  pinSpacer: ".custom-spacer",
  start: "top top",
  end: "+=500"
});
```

or with React:
<details>

```jsx
const triggerRef = useRef();
const pinRef = useRef();
const spacerRef = useRef();

useEffect(() => {
  const trigger = ScrollTrigger.create({
    trigger: triggerRef.current,
    pin: pinRef.current,
    pinSpacer: spacerRef.current,
    start: 'top top',
    end: 'bottom bottom',
  });

  return () => trigger.kill();
}, []);

return (
  <>
    <div ref={spacerRef}>
      <div ref={pinRef}></div>
    </div>
    <div ref={triggerRef}></div>
  </>
);
```
</details>

**Comment by jackdoyle** [comment]
Thanks for the suggestion, @cubeghost. That certainly does sound reasonable. Let me explore that a bit and get back to you. In the meantime, if you've got a minimal demo like in CodeSandbox.io or something like that, please do share.

**Comment by cubeghost** [comment]
Of course! 
Here's a vanilla JS demo of the iframe issue: https://codepen.io/cubeghost/pen/vYmpbaZ
And here's a demo of the React issue: https://codesandbox.io/s/jolly-visvesvaraya-8i7o2?file=/src/App.js

**Comment by OSUblake** [comment]
This seems like it might be really useful feature. 

In the meantime, you should be able to get around the error in React by wrapping your pinned element. 

```jsx
<div>
  <div ref={pinnedRef}>pinned thing</div>
</div>   
```

**Comment by jackdoyle** [comment]
This may be a silly question, but if a wrapper &lt;div&gt; is necessary either way, what's the benefit of this proposed pinSpacer in the API? In other words, how is the second solution better?:

1) Just wrap the pinned element in something (works currently):
```html
<div>
  <div ref={pinnedRef}>pinned thing</div>
</div> 
```

2) Wrap the pinned element in something that we feed to ScrollTrigger as a new "pinSpacer" property (requires update):

```html
<div ref={pinSpacer}>
  <div ref={pinnedRef}>pinned thing</div>
</div> 
```
```javascript
ScrollTrigger.create({
  pinSpacer: pinSpacer.current
  ...
});
```
?

The main difference functionally-speaking is that the pinSpacer in 2 won't get added/removed from the DOM on refresh(). And of course ScrollTrigger wouldn't have to document.createElement("div"). Are either of those significant? Obviously I'm not a React guy, so I may be missing something obvious here.

**Comment by cubeghost** [comment]
yeah, the React issue is definitely secondary here, sorry for the confusion!

**Comment by jackdoyle** [comment]
Nah, that was my forgetfulness. You mentioned the iframe issue initially but I got lost in the weeds with React. 

I've implemented this in the next release which you can preview here: 
- Minified ES5 file: https://assets.codepen.io/16327/ScrollTrigger.min.js
- .tgz file that you can npm install: https://assets.codepen.io/16327/gsap-beta.tgz

Does that work well for you?

**Comment by cubeghost** [comment]
Thanks Jack! We'll try it out ASAP and get back to you

**Comment by cubeghost** [comment]
@jackdoyle this is working great!
